### PR TITLE
[TextServer / Font] Add support for customizable baseline offset.

### LINKED ISSUE #87655: spacing_top/spacing_bottom in fallback FontVariation does not change baseline
### Tested versions

4.2.1.stable

### System information

Linux

### Issue description

When a FontVariation has a FontVariation in it as a fallback font

changing `spacing_glyph` in subresource seems does working after test as #80907 descripted.

But changing `spacing_top` or `spacing_bottom` in subresource does not working.

which I thought changing spacing_bottom/spacing_top will change the position of fallback font, so I can align the fallback font  with main font by baseline.

Screenshot from 2024-01-28 05-07-33

I don't know if this is a bug, or just as it is.
and a new align baseline for fallback font feature need to be implemented.

### Steps to reproduce

   1. Create a Label.
   2.  Create and set a FontVariation as the font of the label node.
   3.  Create and set another FontVariation as a fallback font of the font of the label node.
   4. changing fallback font's spacing_top and spacing_bottom takes no effect. but spacing_glyph works. 

### Minimal reproduction project (MRP)

[bug_font_fallback.zip](https://github.com/godotengine/godot/files/14073835/bug_font_fallback.zip)

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot/issues/87655

https://github.com/godotengine/godot/assets/7645683/598ec12a-abca-41c2-8f79-e6b5dc45a637

### CODE REVIEW

File: scene/resources/font.h [hunk 1]
```diff
@@ -105,6 +105,7 @@ class Font : public Resource {
 
 #ifndef DISABLE_DEPRECATED
 	RID _find_variation_compat_80954(const Dictionary &p_variation_coordinates, int p_face_index = 0, float p_strength = 0.0, Transform2D p_transform = Transform2D()) const;
+	RID _find_variation_compat_xxxxx(const Dictionary &p_variation_coordinates, int p_face_index = 0, float p_strength = 0.0, Transform2D p_transform = Transform2D(), int p_spacing_top = 0, int p_spacing_bottom = 0, int p_spacing_space = 0, int p_spacing_glyph = 0) const;
```
Comment (AThousandShips) [inline scene/resources/font.h:108]:
```suggestion
	RID _find_variation_bind_compat_87668(const Dictionary &p_variation_coordinates, int p_face_index = 0, float p_strength = 0.0, Transform2D p_transform = Transform2D(), int p_spacing_top = 0, int p_spacing_bottom = 0, int p_spacing_space = 0, int p_spacing_glyph = 0) const;
```

File: scene/resources/font.h [hunk 2]
```diff
@@ -118,7 +119,7 @@ class Font : public Resource {
 	virtual TypedArray<Font> get_fallbacks() const;
 
 	// Output.
-	virtual RID find_variation(const Dictionary &p_variation_coordinates, int p_face_index = 0, float p_strength = 0.0, Transform2D p_transform = Transform2D(), int p_spacing_top = 0, int p_spacing_bottom = 0, int p_spacing_space = 0, int p_spacing_glyph = 0) const { return RID(); };
+	virtual RID find_variation(const Dictionary &p_variation_coordinates, int p_face_index = 0, float p_strength = 0.0, Transform2D p_transform = Transform2D(), int p_spacing_top = 0, int p_spacing_bottom = 0, int p_spacing_space = 0, int p_spacing_glyph = 0, float p_baseline_offset = 0.0) const { return RID(); };
```
Comment (AThousandShips) [inline scene/resources/font.h:122]:
```suggestion
	virtual RID find_variation(const Dictionary &p_variation_coordinates, int p_face_index = 0, float p_strength = 0.0, Transform2D p_transform = Transform2D(), int p_spacing_top = 0, int p_spacing_bottom = 0, int p_spacing_space = 0, int p_spacing_glyph = 0, float p_baseline_offset = 0.0) const { return RID(); }
```

**Review by Mickeon — APPROVED** [review]
Class reference is just fine.

**Comment by gu-fan** [comment]
very nice

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**